### ISSUE TITLE
Setting permissions on who can sort rows

### ISSUE THREAD

**Issue by kidereo** [optimistdigital/nova-sortable#37]
Many thanks for the package!

My Nova resources are controlled through policies. Only admins can change resources, the rest just view them. 

With this package, it appears that any user can reorder the rows and it will be persisted in the database, regardless of policy constraints. 

Is there a way to allow only admins to change the order, and the rest would just see that arrangement?

I use Nova 2.11

Update: I uninstalled the latest version and installed v1.4.5 instead. Now the functionality works as I expect it and users who cannot manipulate resources also cannot change the row order. I guess this specific issue just crept in subsequently to v1.4.5.

**Comment by nbayramberdiyev** [comment]
You need to override 2 methods in `OptimistDigital\NovaSortable\Traits\HasSortableRows` trait if you want to allow user-based sorting/reorder: [`OptimistDigital\NovaSortable\Traits\HasSortableRows::getSortability`](https://github.com/optimistdigital/nova-sortable/blob/master/src/Traits/HasSortableRows.php#L9) and [`OptimistDigital\NovaSortable\Traits\HasSortableRows::serializeForIndex`](https://github.com/optimistdigital/nova-sortable/blob/master/src/Traits/HasSortableRows.php#L65).

**Example:** Assume that you have a **Category** resource and you want to allow only admin users to change the order. Your Category resource should look something like this:

```php
<?php

namespace App\Nova;

use Illuminate\Http\Request;
use Laravel\Nova\Fields\ID;
use Laravel\Nova\Fields\Text;
use Laravel\Nova\Http\Requests\NovaRequest;
use OptimistDigital\NovaSortable\Traits\HasSortableRows;

class Category extends Resource
{
    use HasSortableRows;

    /**
     * The model the resource corresponds to.
     *
     * @var string
     */
    public static string $model = \App\Models\Category::class;


    // Your code...


    public static function getSortability(NovaRequest $request)
    {
        if ($request->user()->is_admin) {
            return HasSortableRows::getSortability($request);
        }

        return null;
    }

    public function serializeForIndex(NovaRequest $request, $fields = null)
    {
        $sortability = self::getSortability($request);

        if (! $sortability) {
            return parent::serializeForIndex($request, $fields);
        }

        return array_merge(parent::serializeForIndex($request, $fields), [
            'sortable' => $sortability->sortable,
            'sort_on_index' => $sortability->sortable && !$sortability->sortOnHasMany && !$sortability->sortOnBelongsTo,
            'sort_on_has_many' => $sortability->sortable && $sortability->sortOnHasMany,
            'sort_on_belongs_to' => $sortability->sortable && $sortability->sortOnBelongsTo,
        ]);
    }
}
```

**Comment by Tarpsvo** [comment]
Hi guys! I just released version 1.5.0 that includes the PR from @Naoray. This allows you to enable or disable sorting on a per request basis. There's a section in the README about it as well. I hope this will suffice. Good luck!
